### PR TITLE
Upgrade and use fitbit-sdk-types

### PR DESCRIPTION
**Pull request by SergioMorchon**
I'm playing around with this issue https://github.com/SergioMorchon/fitbit-sdk-types/issues/20 to experience using the Fitbit SDK Types within a real Project.

This approach is the more fine-grained, having:
* The build Fitbit SDK process takes *all* the typings to be able to build the multiple pojects (*app*, *companion*).
* The development process takes precise types depending the target (*app*, *companion*) to avoid importing a module that in fact isn't present there.

So that's the reason why eacch `tsconfig.json` in both *app* and *companion* doesn't inherit from the main one (used by the build process).
Also, this will improve some things:
* Now there is no need to type most of input parameters, because the types are already there.
* You can take advantage of the defined interfaces for the UI elements, as `GraphigElement`, `ImageElement` and so on.
* There is no need to declare things like `fetch` nor it's promise parameters: they're already typed and your coude will be safer (you could be manually casting to something wrong, for example).

### CODE REVIEW

File: app/index.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ messaging.peerSocket.onopen = () => {
   changeScreen(new ZoneGroupScreen(changeScreen));
 };
 
-messaging.peerSocket.onmessage = (evt: any): void => {
+messaging.peerSocket.onmessage = (evt): void => {
```
Comment (SergioMorchon) [inline app/index.ts]: Not `any` anymore: it's [`MessageEvent`](https://github.com/SergioMorchon/fitbit-sdk-types/blob/master/shared/messaging.d.ts#L35).

File: app/screens/zone_group_screen.ts [hunk 2]
```diff
@@ -30,28 +30,28 @@ export default class ZoneGroupScreen extends Screen {
...
-                  getTileInfo: function(index: number): IMenuItem {
+                  getTileInfo: function(index) {
                     return {
                       type: "zone-group-pool",
                       value: "Menu item",
                       index: index
                     };
                   },
-                  configureTile: (function(tile: any, info: IMenuItem): void {
+                  configureTile: (tile, info) => {
```
Comment (SergioMorchon) [inline app/screens/zone_group_screen.ts]: `tile` and `info` have types defined for the [`VirtualTileListDelegate`](https://github.com/SergioMorchon/fitbit-sdk-types/blob/master/device/document/elements.d.ts#L58-L61).

File: app/screens/zone_group_screen.ts [hunk 3]
```diff
@@ -30,28 +30,28 @@ export default class ZoneGroupScreen extends Screen {
...
                     if (info.type === "zone-group-pool") {
                       tile.getElementById("text").text = this.zoneGroups[info.index].name;
                       const touch = tile.getElementById("touch-me");
-                      touch.onclick = (evt: any) => {
+                      touch.onclick = () => {
```
Comment (SergioMorchon) [inline app/screens/zone_group_screen.ts]: With an arrow function now you don't need to `bind` tp `this`, having also *types* _inside_ the function boy. Now it's also type-checked :)

File: companion/albumart.ts [hunk 4]
```diff
@@ -27,12 +25,12 @@ export default function updateAlbumArt(albumArtURI: string) {
 
 function getAlbumArt(albumArtURI: string) {
     console.log("getting Album Art", albumArtURI);
-    fetch(albumArtURI, {}).then((response: any) => {
+    fetch(albumArtURI, {}).then((response) => {
```
Comment (SergioMorchon) [inline companion/albumart.ts]: `fetch` now has it's own types: https://github.com/SergioMorchon/fitbit-sdk-types/blob/master/companion/fetch.d.ts

File: tsconfig.json [hunk 5]
```diff
@@ -1,4 +1,7 @@
 {
   "extends": "./node_modules/@fitbit/sdk/sdk-tsconfig.json",
-  "no-implicit-any": false
```
Comment (SergioMorchon) [inline tsconfig.json]: Was this working? looks like the `compilerOptions` property `noImplicitAny`.

**Comment by andrewjw** [comment]
Thanks for taking the time to do this, it's much appreciated. I worry slightly that because the build process doesn't separate the companion and app types there is a danger an error might slip through the build process. If your editor highlights the mistake though hopefully it's unlikely.

Thanks!

**Pull request merged**